### PR TITLE
feat: Add guide article and update its challenge

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Provides the accompanying guide article for Replace Loops using Recursion.  My original challenge wasn't correct (it wasn't the first n elements, it was the first n + 1), so I corrected that too.

### CODE REVIEW

File: guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: Replace Loops using Recursion
+---
+
+# Replace Loops using Recursion
+
+## Problem explanation
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md]: We have been removing these as the instructions are already in the challenge itself.
```suggestion
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md]:
```suggestion
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Write a recursive function, `sum(arr, n)`, that creates the sum of the first elements (from 0 to `n` inclusive) in an array `arr`.
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md]:
```suggestion
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md]:
```suggestion
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@ojeytonwilliams See the requested changes.  Also, we are only hiding the solutions.  The hints are fine being not hidden.

**Comment by ojeytonwilliams** [comment]
Thanks for the review @RandellDawson.  Let me know if you spot anything else.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion.english.md [hunk 5]
```diff
@@ -7,7 +7,7 @@ videoUrl: 'https://www.freecodecamp.org/news/how-recursion-works-explained-with-
 
 ## Description
 <section id='description'>
-Recursion is the concept that a function can be expressed in terms of itself. To help understand this, start by thinking about the following task: multiply the first <code>n</code> elements in an array to create the product of the elements. Using a <code>for</code> loop, you could do this:
+Recursion is the concept that a function can be expressed in terms of itself. To help understand this, start by thinking about the following task: multiply the first elements (from 0 to <code>n</code> inclusive) in an array to create the product of the elements. Using a <code>for</code> loop, you could do this:
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion.english.md]: After reading the 2nd sentence again (see below), I am not sure what the word "first" is doing there.
>To help understand this, start by thinking about the following task: multiply the **first** elements (from 0 to <code>n</code> inclusive) in an array to create the product of the elements.

**Review by RandellDawson — CHANGES_REQUESTED** [review]
I think the second sentence needs a bit of rewording.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion.english.md (hunk 5 above)
Comment (ojeytonwilliams) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion.english.md]: Good point, none of the firsts make sense any more.  I got rid of them both and tweaked the wording in general.

**Review by RandellDawson — APPROVED** [review]

File: guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    return arr[0];
+  } else {
+    return sum(arr, n - 1) + arr[n];
+  }
+}
+
+```
+
+</details>
+<br>
```
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/basic-javascript/replace-loops-using-recursion/index.md]:
```suggestion
```

**Comment by QuincyLarson** [comment]
@ojeytonwilliams Thanks for your contribution.

**Pull request merged**